### ISSUE TITLE
nonFinal, do we still need it?

### ISSUE THREAD

**Issue by arjantijms** [eclipse-ee4j/glassfish-spec-version-maven-plugin#23]
The plug-in uses a convention for something called "nonFinal", which I think didn't carry over to Jakarta:

e.g.

```
//  OSGi Bundle-SymbolicName:   ${API_PACKAGE}-api
//  OSGi bundle specversion:    ${SPEC_VERSION}.99.b${SPEC_BUILD}
//  OSGi Bundle-Version:        ${SPEC_VERSION}.99.b${SPEC_BUILD}
//  jar Extension-Name:         ${API_PACKAGE}
//  jar Specification-Version:  ${SPEC_VERSION}.99.${SPEC_BUILD}
//  jar Implementation-Version: ${NEW_SPEC_VERSION}-b${SPEC_BUILD}
```

Do we need to retain this?

@bshannon @romain-grecourt

**Comment by romain-grecourt** [comment]
The versioning [rules](https://wiki.eclipse.org/JakartaEE_Maven_Versioning_Rules) still mentions non final.

**Comment by arjantijms** [comment]
> The versioning rules still mentions non final.

You're right, good find ;) Though is this "officially" adopted? We may also need an update here anyway for the -SNAPSHOT and -RCx cases.

One small thing, the option is now called "nonFinal", which means we often have to say `"nonFinal" = false`. Double negatives are in general hard for people to get right at a glance.

I know there's some history here, but going forward, can't we just call it `final` with values `true` and `false`?

**Comment by bshannon** [comment]
I think the intent was that final was the default if nothing was said so you would
only ever say nonFinal=true.

**Comment by romain-grecourt** [comment]
I do agree that double negations are hard to understand. A rename to `final` seems like a good future improvement.
